render/fx_renderer/gles2/shaders/tex.frag

```diff
@@ -55,8 +59,26 @@ void main() {
 		return;
 	}
 
-	if (!has_titlebar || gl_FragCoord.y - position.y > radius) {
+	if (round_top_left || round_top_right || round_bottom_left ||
+		round_bottom_right) {
+		vec2 relative_pos = (gl_FragCoord.xy - position);
+		vec2 size = half_size * 2.0;
+
+		// Brachless baby!
+		// Selectively round individual corners
+		float top = step(relative_pos.y - radius, 0.0);
+		float left = step(relative_pos.x - radius, 0.0);
+		float right = step(size.x - radius, relative_pos.x);
+		float bottom = step(size.y - radius, relative_pos.y);
+
+		float top_left = top * left * float(round_top_left);
+		float top_right = top * right * float(round_top_right);
+		float bottom_left = bottom * left * float(round_bottom_left);
+		float bottom_right = bottom * right * float(round_bottom_right);
+		// A value of 1 means that we're allowed to round at this position
+		float round_corners = top_left + top_right + bottom_left + bottom_right;
```

Comment:
lets move this to a function in its own file (similar to how we do for the gradient shader and how we will handle the round rect SDF) - then we can share it between this shader and the round rect shader